### PR TITLE
PLT-6832 Allow params to be empty for some JSON-RPC methods

### PR DESCRIPTION
**Pull request by neilmayhew**
* `getCurrentSyncedBlock` method
* `getTargetAddresses` method
* Update documentation
* Improve error reporting in `marconi-sidechain/examples/run-pre-prod-queries`

Note that we allow `params` to be empty but not missing, because we don't want `params` to be optional for other methods and they all share a common `Request p` datatype.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [x] Commit sequence broadly makes sense and have useful messages
    - [ ] Important changes are reflected in changelog.d of the affected packages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [x] Targeting main unless this is a cherry-pick backport
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [ ] If relevant, reference the ADR in the PR and reference the PR in the ADR
    - [x] Reviewer requested

### CODE REVIEW

File: marconi-sidechain/doc/API.adoc [hunk 1]
```diff
@@ -245,7 +258,7 @@ Taking the chainpoint before ensure that we have consistent infomation across al
 
 ==== Pre-conditions
 
-None as there are no required inputs.
+None as the input parameter is unused.
```
Comment (berewt) [inline marconi-sidechain/doc/API.adoc:261]: Now that I have a closer look at it, isn't both this version and the original one inaccurate? There is indeed a precondition on the parameter, as we don't accept arbitrary values.

File: marconi-sidechain/src/Marconi/Sidechain/Bootstrap.hs
Comment (berewt) [inline marconi-sidechain/src/Marconi/Sidechain/Bootstrap.hs:1]: Just be careful with the last merge/rebase as it seems that you used your other branch as a base for this PR.

File: marconi-chain-index/json-rpc/src/Network/JsonRpc/Types.hs [hunk 2]
```diff
@@ -177,6 +178,17 @@ instance (ToJSON e, ToJSON r) => ToJSON (JsonRpcResponse e r) where
...
+
+instance FromJSON UnusedRequestParams where
+  parseJSON Null = pure UnusedRequestParams
+  parseJSON (String "") = pure UnusedRequestParams
+  parseJSON (Object o) | null o = pure UnusedRequestParams
+  parseJSON _ = fail "The param value must be empty (use '{}', 'null' or empty string)"
+
+instance ToJSON UnusedRequestParams where
+  toJSON = const Null
+
```
Comment (berewt) [inline marconi-chain-index/json-rpc/src/Network/JsonRpc/Types.hs:191]: 👍

**Review by berewt — APPROVED** [review]
Just a documentation point, if you agree with this one, can you see if we have a similar precondition problem elsewhere?

File: marconi-sidechain/doc/API.adoc (hunk 1 above)
Comment (neilmayhew) [reply marconi-sidechain/doc/API.adoc:261]: Good catch. I'll change it. This applies to both `getCurrentSyncedBlock` and `getTargetAddresses`.

**Pull request merged**